api/src/main/scala/za/co/absa/loginsvc/rest/config/auth/ActiveDirectoryLDAPConfig.scala

```diff
@@ -57,3 +72,110 @@ case class ActiveDirectoryLDAPConfig(domain: String, url: String, searchFilter:
     else ConfigValidationSuccess
   }
 }
+
+case class ServiceAccountConfig(private val accountPattern: String,
+                                private val inConfigAccount: Option[IntegratedLdapUserConfig],
+                                private val awsSecretsManagerAccount: Option[AwsSecretsLdapUserConfig])
+{
+  private val ldapUserDetails: LdapUser = (inConfigAccount, awsSecretsManagerAccount) match {
+    case (Some(_), Some(_)) =>
+      throw ConfigValidationException("Both integratedLdapUserConfig and awsSecretsLdapUserConfig exist. Please choose only one.")
+
+    case (None, None) =>
+      throw ConfigValidationException("Neither integratedLdapUserConfig nor awsSecretsLdapUserConfig exists. One of them should exist.")
+
+    case _ =>
+      val ldapConfig: LdapUser = inConfigAccount.orElse(awsSecretsManagerAccount).get
+      ldapConfig.throwErrors()
+
+      ldapConfig
```

Comment:
```suggestion
    case (inConfig@Some(_), None) | (None, awsConfig@Some(_)) =>
      val ldapConfig: LdapUser = inConfig.orElse(awsConfig).get
      ldapConfig.throwOnErrors()
```

Maybe make use of the matching, callers could still pass e.g. `null`s there.

Reply:
Done